feat: consolidate settings gui

Kuchenpirat opened the pull request:
## What this PR does / why we need it:

Settings are somewhat hidden behind the user avatar in the side bar and take up a lot of space at the bottom for not frequently changed settings. 

This consolidates all settings behind an universal settings button with a menu. 

## Which issue(s) this PR fixes:

None, just some cleaning up. 

## Special notes for your reviewer:

Had to update some test. Not really familliar with playwright but it seems to be working as expected now. 

## Testing

Manual

## Screenshot
| Before | After | Mobile (After) |
| -- | -- | -- |
| <img width="263" height="211" alt="image" src="https://github.com/user-attachments/assets/d61038df-6859-4647-9d63-56beb08e6d66" />  | <img width="486" height="282" alt="image" src="https://github.com/user-attachments/assets/f9b5aea8-df4c-4dac-96b6-39533fbc2aae" /> |  <img width="362" height="261" alt="image" src="https://github.com/user-attachments/assets/cd36e67e-ea2f-44a4-a723-1f8aabfb6a4f" /> |

Kuchenpirat committed:
consolidate settings

Kuchenpirat committed:
fix tests?

Kuchenpirat committed:
Merge branch 'mealie-next' into consolidate-settings

Kuchenpirat committed:
fix tests v2?

Kuchenpirat committed:
fix test v3?

michael-genson committed:
Merge branch 'mealie-next' into consolidate-settings

michael-genson reviewed (changes requested):
Love this. Just a couple suggestions.

Can we change "settings" to something else, like "user settings"? Seems strange to click on "settings" and then click on "settings" again.
<img width="468" height="237" alt="image" src="https://github.com/user-attachments/assets/3671d583-851c-4761-869d-a79b67f21843" />

It also might be worth adding a "data management" option and linking directly to the data management page (which defaults to food management). Then we could also remove it from the user profile page. Honestly a button for user/household/group settings might not be a bad idea here and then we can split up the user profile settings page.

Also, would it be better to leave the language selector out of this menu? I'm thinking if it's a user's first time opening the site and they don't speak/read English they might have a hard time finding the language setting. Not sure if this is necessary but just a thought I had.

Kuchenpirat commented:
Yeah, the naming does make sense 👍
Direct link to data management does also make sense to me 👍
I have to make a version with all the options to have this visualized, might be a bit too much, but I'll try it out. Like the idea in principle. 

Mealie defaults to the browser language I think, so people should usually not need to change language and it's loads cleaner to just have one button. Also I think it's pretty standard to have the language button behind another button afaik.

michael-genson commented:
> I have to make a version with all the options to have this visualized, might be a bit too much

For sure, if it ends out being too cluttered then how you have it is fine (and we can always revisit if we come up with something better)

> Also I think it's pretty standard to have the language button behind another button afaik

Sounds good to me, I'm spoiled by English-default so it's not something I'm terribly familiar with.

Kuchenpirat committed:
use existing admin settings translation

Kuchenpirat committed:
rename to user settings

Kuchenpirat committed:
found existing user settings :D

Kuchenpirat committed:
Add Manage Data Link to Sidebar

Kuchenpirat committed:
Merge branch 'mealie-next' into consolidate-settings

Kuchenpirat commented:
Updated the language and added the manage data links. 

Seperating out the settings is a bit to much for this layout imho, so left it out for now. 

Also verified that language defaults to the system or browser language so having it behind a settings menu should be no problem, especially since the settings menu and the languge button have icons that should make it easy even if you dont understand the currently displayed language.

Kuchenpirat committed:
Merge branch 'mealie-next' into consolidate-settings

michael-genson commented on frontend/components/Layout/LayoutParts/AppSidebar.vue:
```diff
@@ -82,20 +83,22 @@
+            <v-list-item :prepend-icon="$vuetify.theme.current.dark ? $globals.icons.weatherSunny : $globals.icons.weatherNight" :title="$vuetify.theme.current.dark ? $t('settings.theme.light-mode') : $t('settings.theme.dark-mode')" @click="toggleDark" />
+            <v-divider v-if="loggedIn" class="my-2" />
+            <v-list-item v-if="loggedIn" :prepend-icon="$globals.icons.cog" :title="$t('profile.user-settings')" to="/user/profile" />
+            <v-list-item v-if="canManage" :prepend-icon="$globals.icons.manageData" :title="$t('data-pages.data-management')" to="/user/profile" />
```
```suggestion
            <v-list-item v-if="canManage" :prepend-icon="$globals.icons.manageData" :title="$t('data-pages.data-management')" to="/group/data" />
```

michael-genson reviewed (changes requested):
LGTM, save for one issue with the new data management link

Kuchenpirat committed:
Update frontend/components/Layout/LayoutParts/AppSidebar.vue

Co-authored-by: Michael Genson <71845777+michael-genson@users.noreply.github.com>

Kuchenpirat committed:
Merge branch 'mealie-next' into consolidate-settings

michael-genson reviewed (approved):
LGTM

Kuchenpirat committed:
Merge branch 'mealie-next' into consolidate-settings
